src/main/java/io/logz/jmx2logzio/configuration/Jmx2LogzioConfiguration.java

```diff
@@ -0,0 +1,228 @@
+
+            // Try to find hostname as default to serviceHost in case it was not provided
+            if (serviceHost == null) {
+                try {
+                    serviceHost = InetAddress.getLocalHost().getHostName();
+                } catch (UnknownHostException e) {
+                    throw new IllegalConfiguration("service.host was not defined, and could not determine it from the servers hostname");
+                }
+            }
+        }
+        try {
+            whiteListPattern = Pattern.compile(config.hasPath(Jmx2LogzioJavaAgent.WHITE_LIST_REGEX) ?
+                    config.getString(Jmx2LogzioJavaAgent.WHITE_LIST_REGEX) : ".*");
+        } catch (Exception e) {
+            logger.error("Failed to parse regex {} with error {}", config.getString(Jmx2LogzioJavaAgent.WHITE_LIST_REGEX), e.getMessage());
+            whiteListPattern = Pattern.compile(".*");
+        }
+
+        try { // $a is a regexp that will never match anything (will match an "a" character after the end of the string
+            blackListPattern = Pattern.compile(config.hasPath(Jmx2LogzioJavaAgent.BLACK_LIST_REGEX) ?
+                    config.getString(Jmx2LogzioJavaAgent.BLACK_LIST_REGEX) : "$a");
+        } catch (Exception e) {
+            logger.error("Failed to parse regex {} with error {}", config.getString(Jmx2LogzioJavaAgent.WHITE_LIST_REGEX), e.getMessage());
+            blackListPattern = Pattern.compile("$a");
+        }
+
+        serviceName = config.getString(Jmx2LogzioJavaAgent.SERVICE_NAME);
+
+        logzioJavaSenderParams = new LogzioJavaSenderParams();
+        if (config.hasPath(LogzioJavaSenderParams.LISTENER_URL)) {
+            URL url = null;
+            try {
+                url = new URL(config.getString(LogzioJavaSenderParams.LISTENER_URL));
+                logzioJavaSenderParams.setUrl(url.toString());
+            } catch (MalformedURLException e) {
+                logger.error("malformed listener URL {} got error {}. Using default listener URL: {}",
+                        config.getString(LogzioJavaSenderParams.LISTENER_URL),e.getMessage(),logzioJavaSenderParams.getUrl());
+            }
+        }
+        if (config.getString(LogzioJavaSenderParams.LOGZIO_TOKEN).equals("<your_logz.io_token>")) {
+            throw new IllegalConfiguration("please enter a valid logz.io token (can be located at apps.logz.io >> settings >> general)");
+        }
+        logzioJavaSenderParams.setToken(config.getString(LogzioJavaSenderParams.LOGZIO_TOKEN));
+
+        logzioJavaSenderParams.setFromDisk(config.hasPath(LogzioJavaSenderParams.FROM_DISK) ?
+                config.getBoolean(LogzioJavaSenderParams.FROM_DISK) : logzioJavaSenderParams.isFromDisk());
+
+        if (config.hasPath(LogzioJavaSenderParams.IN_MEMORY_QUEUE_CAPACITY)) {
+            int capacity = config.getInt(LogzioJavaSenderParams.IN_MEMORY_QUEUE_CAPACITY);
+            if (capacity > 0) {
```

Comment:
trinary condition

Reply:
What's the motivation behind this? this isn't the traditional form of:
`if (expression) then
    var = a
  else 
    var = b `
if we'll use ternary condition it will look something like this:
`var = expression ? a : var`
which will cause a redundant placement.